applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+
+        // check the scheme
+        this->pGetScheme()->Check(rModelPart);
+
+        // check the builder and solver
+        this->pGetBuilderAndSolver()->Check(rModelPart);
+
+        KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            << "Exiting Check" << std::endl;
+
+        return 0;
+
+        KRATOS_CATCH("")
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    SchemePointerType mpScheme;
+
+    BuilderAndSolverPointerType mpEigenSolver;
+
+    BuilderAndSolverPointerType mpBuilderAndSolver;
+
+    // SparseMatrixPointerType mpMassMatrix;
+
+    SparseMatrixPointerType mpStiffnessMatrix;
+
+    SparseMatrixPointerType mpStiffnessMatrixPrevious;
+
+    SparseVectorPointerType mpRHS;
+
+    SparseVectorPointerType mpDx;
+
+    typename ConvergenceCriteriaType::Pointer mpConvergenceCriteria;
+    /// The pointer to the convergence criteria employed
+
+    bool mInitializeWasPerformed = false;
+
+    bool mSolutionStepIsInitialized = false;
+
+    bool mSolutionFound = false;
+
+    unsigned int mLoadStepIteration = 0;
+
+    unsigned int mMaxIteration;
+
+    std::vector<array_1d<double,3>> mpInitialLoads;
+
+    double mInitialStep;
+    double mSmallStep;
+    double mBigStep;
+    double mConvergenceRatio;
+
+    double lambda = 0.0;
+    double lambdaPrev = 1.0;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+/*
+Store Initial Loads to ensure that load multiplier always refers to user input
+*/
+    void StoreInitialLoadConditions()
+    {
+        ModelPart &rModelPart = BaseType::GetModelPart();
+        const auto it_conditions_begin = rModelPart.ConditionsBegin();
+        for (int j = 0; j < static_cast<int>(rModelPart.Conditions().size()); j++) 
+        {
+            auto it_condition = it_conditions_begin + j;
+            // Check if condition is point load
+            if (it_condition->Has(POINT_LOAD))
+            {
+                mpInitialLoads.push_back( it_condition->GetValue(POINT_LOAD) );
+            }
+            // Check if condition is surface load
+            if (it_condition->Has(SURFACE_LOAD))
+            {
+                mpInitialLoads.push_back( it_condition->GetValue(SURFACE_LOAD) );
+            }
+        } 
+    }
+/*
```

Comment:
Wrong spaces and missing *
```suggestion
/**
```